### ISSUE TITLE
disable page refresh

### ISSUE THREAD

**Issue by sydcanem** [atom/electron#3725]
Is there any way to disable the app refreshing when using shortcuts? I can't seem to find any documentation on this.

**Comment by zcbenz** [comment]
You can use the `will-navigate` event.
http://electron.atom.io/docs/v0.35.0/api/web-contents/#event-39-will-navigate-39

**Comment by pjebs** [comment]
This does not seem to work anymore!

**Comment by Christopher-Hayes** [comment]
I haven't been able to capture the reload event with `will-navigate`; however, `did-start-navigation` does capture the refresh event for me. It probably depends on the type of app you're making. For me checking if the `isInPlace` argument is `false` would predictably indicate if the refresh was triggered. However, `did-start-navigation` does not allow you to stop the navigation.

Honestly, the easiest solution is to change the [`Cmd/Ctrl+R` electron shortcut](https://electronjs.org/docs/api/accelerator).

**TL;DR**

    const { app, globalShortcut } = require('electron');

    app.on('ready', async () => {
        globalShortcut.register('CmdOrCtrl+R', () => {});
    });

*Remember to change the Electron menu to omit the page refresh menu item*

**Comment by tothradoslav** [comment]
@Christopher-Hayes  this works, but it's not the solution, since this stops CMD+R from working ALL apps in OS (using OSX). I can't even reload my chrome page anymore :)

Does anyone have a real solution to altering the CMD+R behavior and catching the event?

**Comment by pjebs** [comment]
See https://electron.guide

**Comment by tothradoslav** [comment]
@pjebs Thank you, saved my week!

**Comment by xieerduos** [comment]
```js   
function createWindow() {
    mainWindow = new BrowserWindow({
        width: 900,
        height: 620,
        show: false,
        modal: true,
        backgroundColor: '#fff',
        //  Boolean(可选)-对 Windows 上的无框窗口使用WS_THICKFRAME 样式，会增加标准窗口框架。
        // 设置为 false 时将移除窗口的阴影和动画. 默认值为 true。
        thickFrame: false
    });    

    mainWindow.on('ready-to-show', () => {
        mainWindow.show();
        // 删除窗口的菜单栏。
        mainWindow.removeMenu();
   });
}

OR 

function createWindow() {
   mainWindow = new BrowserWindow({
        width: 900,
        height: 620
    });    

Menu.setApplicationMenu(Menu.buildFromTemplate([]))

}
```

**Comment by xzifan** [comment]
in most of the cases you need to disable the `cmd + r` shortcut on a target browser window. 
use `before-input-event` on webContents.

```js
webContents.on("before-input-event", (event, input) => {
    // disable refresh
    if ((input.control || input.meta) && input.key.toLowerCase() === 'r') {
      event.preventDefault();
    }
    // disable zoom
    if ((input.control || input.meta) && (input.key.toLowerCase() === '-' || input.key.toLowerCase() === '=')) {
      e.preventDefault();
    }
});
```

> https://stackoverflow.com/questions/55296837/how-to-disable-keyboard-shortcuts-alt-f4-in-electronjs

if you want to disable shortcut on system level, use `globalShortcut` from `electron`
